scaaml/capture/scope.py

```diff
@@ -0,0 +1,54 @@
+"""Context manager for the scope."""
+
+from abc import ABC, abstractmethod
+
+
+class AbstractSScope(ABC):
+    """Scope context manager."""
+    def __init__(self, gain: int, samples: int, offset: int, clock: int, **_):
+        """Create scope context.
+
+        Args:
+          gain: Gain of the scope.
```

Comment:
`gain` and `clock` are specific to Chipwhisperer. A more traditional scope wouldn't have such controls.
Normally you deal with `samplerate` (how many samples per second you want to capture), `timebase` expressed in seconds (or ms, us, ns, etc.) per division (a typical scope screen has 10 divisions and the timebase is the horizontal resolution), `vertical scale` expressed in volts per division (a typical scope has 10 divisions).

So `samples` would be `timebase * 10 * samplerate`, `gain` is the vertical scale but with Chipwhisperer it's the amplification factor in dB, and `clock` is related to the `samplerate` as the Chipwhisperer can acquire either at `1x` or `4x` of this clock speed.

So if you really want an abstract class, it shouldn't be tainted by Chipwhisperer oddities IMO :)